db/migrate/20170612115845_dedupe_embedded_specialist_change_history.rb

```diff
@@ -0,0 +1,102 @@
+class DedupeEmbeddedSpecialistChangeHistory < ActiveRecord::Migration[5.1]
+  disable_ddl_transaction!
+
+  def up
+    # Find ChangeNotes which have no edition id and
+    # have duplicate documents / notes, these need to be removed first.
+    # The only publishing apps affected are specialist publisher and service manual publisher.
+    dupes_sql = <<-SQL
+      SELECT max(id)
+      FROM change_notes
+      WHERE edition_id IS NULL
+      GROUP BY document_id, note
+      HAVING (count(*) > 1)
```

Comment:
I'm a bit confused by this query and the loop - maybe I'm totally missing something. I'm assuming we're pulling out one ID each time, if so why not all in one, and/or why the max one?